src/tools/gophish_complete.py

```diff
@@ -137,35 +159,60 @@ def main():
         )
     except ValueError:
         logging.critical(
-            '"{}"is not a valid logging level.  Possible values are debug, info, warning, and error.'.format(
-                log_level
-            )
+            f'"{log_level}"is not a valid logging level.  Possible values are debug, info, warning, and error.'
         )
         return 1
 
+    # Connect to API
+    try:
+        api = connect_api(args["API_KEY"], args["SERVER"])
+        logging.debug(f'Connected to: {args["SERVER"]}')
+    except Exception as e:
+        logging.critical(print(e.args[0]))
+        sys.exit(1)
+
+    # Gets assessment id from campaign name or user input.
+    if not args["CAMPAIGN_NAME"]:
+        assessment_id = get_input("Enter the Assessment ID")
     else:
-        # Connect to API
-        try:
-            api = connect_api(args["API_KEY"], args["SERVER"])
-            logging.debug("Connected to: {}".format(args["SERVER"]))
-        except Exception as e:
-            logging.critical(print(e.args[0]))
-            sys.exit(1)
-
-    if args["--complete"]:
-        campaigns = get_campaigns(api)
-        workingID = select_campaigns(campaigns)
-        success = complete_campaign(api, args["API_KEY"], args["SERVER"], workingID)
-
-    elif args["--summary"]:
-        campaigns = get_campaigns(api)
-        workingID = select_campaigns(campaigns)
-        success = print_summary(api, workingID)
-
-    elif args["--auto"]:
-        success = complete_campaign(
-            api, args["API_KEY"], args["SERVER"], args["--auto"]
-        )
+        # Sets assessment id from first section of campaign name. If the
+        # assessment wizard is used to build the assessment the campaign
+        # name will allways start with the assessment identifier.
+        assessment_id = args["CAMPAIGN_NAME"].split("-")[0]
+
+    # Gather all campaigns associated with assessment identifier.
```

Comment:
If the campaign name is provided, why do we need to derive the assessment_id and fetch all campaigns from that assessment?  Can't we just find the requested campaign directly from a dict of all campaigns?

Reply:
Good point, and I think this would be achievable. If we provide a default value of `null` for `assessment_id`, then I believe GoPhish would return all campaigns. I will test this and make the required changes.

Reply:
Made this change in commit 59ca3fb. The script sets the `assessment_id` to a blank string so all campaigns will be returned and search for the campaign name.